packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -258,30 +342,38 @@ export const useUserProfileContext = (): UserProfileContextType => {
 
 export type UserVerificationContextType = UserVerificationCtx;
 
+export const UserVerificationContext = React.createContext<UserVerificationCtx | null>(null);
+
 export const useUserVerification = (): UserVerificationContextType => {
-  const { componentName, ...ctx } = (React.useContext(ComponentContext) || {}) as UserVerificationCtx;
+  const context = React.useContext(UserVerificationContext);
 
-  if (componentName !== 'UserVerification') {
+  if (!context || context.componentName !== 'UserVerification') {
     throw new Error('Clerk: useUserVerificationContext called outside of the mounted UserVerification component.');
   }
 
+  const { componentName, ...ctx } = context;
+
   return {
     ...ctx,
     componentName,
   };
```

Comment:
Looks like this could be simplified to just 
```suggestion
  return context;
```

Reply:
yeah this is left over from the previous implementation. When I refactor this to split these contexts into separate files and remove the `componentName` check I'll clean this up as well!